nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -97,6 +97,15 @@ while [ "$#" -gt 0 ]; do
 done
 
 
+# if necessary, request elevated privileges early on
+if [ "$action" = switch -o "$action" = boot -o "$action" = test -o "$upgrade" = 1 ]; then
+       if [ $EUID != 0 ]; then
+               sudo "$0" "$origArgs"
```

Comment:
Not all systems have sudo, please consider falling back to su.

Reply:
Fixed.